sensor/common/networkflow/manager/manager_impl_test.go

```diff
@@ -179,7 +193,521 @@ func (suite *NetworkflowManagerTestSuite) TestAddNoOriginator() {
 	h.connectionsSequenceID = sequenceID
 
 	err := h.Process(networkInfoOpen, nowTimestamp, sequenceID)
-	suite.NoError(err)
+	s.NoError(err)
+
+	s.Len(h.endpoints, 1)
+}
+
+//endregion
+
+//region networkFlowManager tests
+
+func (s *NetworkFlowManagerTestSuite) TestEnrichConnection() {
+	mockCtrl := gomock.NewController(s.T())
+	m, mockEntityStore, mockExternalSrc, _ := createManager(mockCtrl)
+	containerID := "id"
+	cases := map[string]struct {
+		connPair                    *connectionPair
+		enrichConnections           map[networkConnIndicator]timestamp.MicroTS
+		expectEntityLookupContainer expectFn
+		expectEntityLookupEndpoint  expectFn
+		expectExternalLookup        expectFn
+		expectedIndicator           *networkConnIndicator
+		expectedConnection          *connection
+		expectedStatus              *connStatus
+	}{
+		"Rotten connection": {
+			connPair:                    createConnectionPair(true, true, false, timestamp.Now().Add(-maxContainerResolutionWaitPeriod*2)),
+			expectEntityLookupContainer: expectEntityLookupContainerHelper(mockEntityStore, 1, clusterentities.ContainerMetadata{}, false),
+			expectedStatus: &connStatus{
+				rotten: true,
+			},
+		},
+		"Incoming External connection no external sources hit": {
+			connPair:          createConnectionPair(true, true, false, timestamp.Now()),
+			enrichConnections: make(map[networkConnIndicator]timestamp.MicroTS),
+			expectEntityLookupContainer: expectEntityLookupContainerHelper(mockEntityStore, 1, clusterentities.ContainerMetadata{
+				DeploymentID: containerID,
+			}, true),
+			expectExternalLookup: expectExternalLookupHelper(mockExternalSrc, 1, nil),
+			expectedStatus: &connStatus{
+				used: true,
+			},
+			expectedIndicator: &networkConnIndicator{
+				dstPort:   80,
+				protocol:  net.TCP.ToProtobuf(),
+				srcEntity: networkgraph.InternetEntity(),
+				dstEntity: networkgraph.EntityForDeployment(containerID),
+			},
+		},
+		"Outgoing External external sources hit": {
+			connPair:          createConnectionPair(false, true, false, timestamp.Now()),
+			enrichConnections: make(map[networkConnIndicator]timestamp.MicroTS),
+			expectEntityLookupContainer: expectEntityLookupContainerHelper(mockEntityStore, 1, clusterentities.ContainerMetadata{
+				DeploymentID: containerID,
+			}, true),
+			expectExternalLookup: expectExternalLookupHelper(mockExternalSrc, 1, &storage.NetworkEntityInfo{
+				Id: containerID,
+			}),
+			expectedStatus: &connStatus{
+				used: true,
+			},
+			expectedIndicator: &networkConnIndicator{
+				dstPort:  80,
+				protocol: net.TCP.ToProtobuf(),
+				dstEntity: networkgraph.EntityFromProto(&storage.NetworkEntityInfo{
+					Id: containerID,
+				}),
+				srcEntity: networkgraph.EntityForDeployment(containerID),
+			},
+		},
+		"Incoming": {
```

Comment:
I think, it may be beneficial to rewrite the test case names, so that it includes the word `should`. It is not an issue when the names are long, but at least we would know what a given case is testing.

Reply:
I've renamed pretty much all the cases in `TestEnrichConnection`, `TestEnrichContainerEndpoint`, and `TestEnrichProcessListening`. Let me know if you think they are more understandable now.

Reply:
Thanks for that! I feel the difference is huge :)